pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -190,6 +200,12 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		if err != nil {
 			// continue execution to allow other maintenance activities to continue
 			workerLog.Error(err, "Could not maintain Kubernetes version for worker pool")
+		} else {
+			if maintainedShoot.Spec.Provider.Workers[i].Kubernetes != nil && maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Kubelet != nil {
+				if reasons := maintainFeatureGates(&maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Kubelet.KubernetesConfig, *maintainedShoot.Spec.Provider.Workers[i].Kubernetes.Version, fmt.Sprintf("spec.provider.workers[%d].kubernetes.kubelet.featureGates", i)); len(reasons) > 0 {
```

Comment:
I am not sure if the assumption that if worker pool specific has `kubernetes.kubelet.featureGates` set, then it will also have the worker pool specific `kubernetes.version`. The worker pool specific field `kubernetes.version` is optional, hence you could only have `kubernetes.kubelet.featureGates` set for a worker pool.
https://github.com/gardener/gardener/blob/b33edc4fb1fd2723116510245a7ab77b82ab7f3a/pkg/apis/core/v1beta1/types_shoot.go#L1499-L1511: both fields are optional. I think there was a helper func to determine the K8s version of a worker pool.

Reply:
If the kubernetes version has not changed, then the `maintainFeatureGates` will be just no-op, right?